contracts/nexus-gateway/src/msg.rs

```diff
@@ -1,17 +1,26 @@
 use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::HexBinary;
 use msgs_derive::EnsurePermissions;
+use router_api::{Address, ChainName};
 
 use crate::nexus;
 
 #[cw_serde]
 pub struct InstantiateMsg {
     pub nexus: String,
     pub router: String,
+    pub axelar_gateway: String,
 }
 
 #[cw_serde]
 #[derive(EnsurePermissions)]
 pub enum ExecuteMsg {
+    #[permission(Any)]
+    CallContractWithToken {
+        destination_chain: ChainName,
+        destination_address: Address,
+        payload: HexBinary,
+    },
```

Comment:
we should add a comment saying this only works for legacy chains. Also, what is the behavior if the destination chain is not known to core? Will the caller receive an error?

Reply:
The tx would fail. Well, everything in the nexus gateway works only for the legacy chains, not just this msg type.

Reply:
yeah would be good to add some comments so that this is clear